src/web_app/js/peerconnectionclient.js

```diff
@@ -360,5 +375,215 @@ PeerConnectionClient.prototype.onRemoteStreamAdded_ = function(event) {
+      localVideoTagId);
+    }
+  }
+
+  if (this.pc_.getLocalStreams()[0].getAudioTracks().length > 0) {
+    // Determine local audio track id, label and SSRC.
+    var audioSendTrackId =
+        this.pc_.getLocalStreams()[0].getAudioTracks()[0].id;
+    var audioSendTrackLabel =
+        this.pc_.getLocalStreams()[0].getAudioTracks()[0].label;
+    var audioSendSsrcLine = this.pc_.localDescription.sdp.match('a=ssrc:.*.' +
+        audioSendTrackId);
+
+    // Firefox does not use SSRC.
+    if (audioSendSsrcLine !== null) {
+      var audioSendSsrc = audioSendSsrcLine[0].match('[0-9]+');
+      this.callstats.associateMstWithUserID(this.pc_, this.userId,
+          this.conferenceId, audioSendSsrc[0], audioSendTrackLabel,
+          localVideoTagId);
+    }
+  }
+};
+
+// Associate receive SSRC's with media stream tracks and user ID in the
+// callstats backend.
+PeerConnectionClient.prototype.bindReceiveMstToIdForCallstats_ = function() {
+  if (!this.callstats && this.pc_.getRemoteStreams().length === 0 &&
+      typeof this.pc_.remoteDescription.sdp === 'undefined') {
+    trace('Cannot associate receive mst with userId.');
+    return;
+  }
+
+  var remoteVideoTagId = 'remote-video';
+
+  if (this.pc_.getRemoteStreams()[0].getVideoTracks.length > 0) {
+    // Determine remote video track id, label and SSRC.
+    var videoReceiveTrackId =
+        this.pc_.getRemoteStreams()[0].getVideoTracks()[0].id;
+    var videoReceiveTrackLabel =
+        this.pc_.getRemoteStreams()[0].getVideoTracks()[0].label;
+    var videoReceiveSsrcLine =
+        this.pc_.remoteDescription.sdp.match('a=ssrc:.*.' +
+        videoReceiveTrackId);
+
+    // Firefox does not use SSRC.
+    if (videoReceiveSsrcLine !== null) {
+      var videoReceiveSsrc = videoReceiveSsrcLine[0].match('[0-9]+');
+      this.callstats.associateMstWithUserID(this.pc_, this.remoteUserId,
+          this.conferenceId, videoReceiveSsrc[0], videoReceiveTrackLabel,
+          remoteVideoTagId);
+    }
+  }
+
+  if (this.pc_.getRemoteStreams()[0].getAudioTracks.length > 0) {
+    // Determine remote audio track id, label and SSRC.
+    var audioReceiveTrackId =
+        this.pc_.getRemoteStreams()[0].getAudioTracks()[0].id;
+    var audioReceiveTrackLabel =
+        this.pc_.getRemoteStreams()[0].getAudioTracks()[0].label;
+    var audioReceiveSsrcLine =
+        this.pc_.remoteDescription.sdp.match('a=ssrc:.*.' +
+        audioReceiveTrackId);
+
+    // Firefox does not use SSRC.
+    if (audioReceiveSsrcLine !== null) {
+      var audioReceiveSsrc = audioReceiveSsrcLine[0].match('[0-9]+');
+      this.callstats.associateMstWithUserID(this.pc_, this.remoteUserId,
+          this.conferenceId, audioReceiveSsrc[0], audioReceiveTrackLabel,
+          remoteVideoTagId);
+    }
+  }
+};
+
+// Send events to callstats backend.
+PeerConnectionClient.prototype.sendCallstatsEvents = function(fabricEvent) {
+  // http://www.callstats.io/api/#enumeration-of-fabricevent
+  var fabricEvents = {
```

Comment:
why not just pass through the right events from the start, instead of translating them here? Or maybe create our own constants?